src/extension/background-script/connectors/alby.ts

```diff
@@ -285,28 +283,60 @@ export default class Alby implements Connector {
       });
 
       authUrl += "&webln=false"; // stop getalby.com login modal launching lnurl auth
-      const authResult = await this.launchWebAuthFlow(authUrl);
-      const code = new URL(authResult).searchParams.get("code");
-      if (!code) {
-        throw new Error("Authentication failed: missing authResult");
-      }
 
-      const token = await authClient.requestAccessToken(code);
-      await this._updateOAuthToken(token.token);
-      return authClient;
+      const oAuthTab = await browser.tabs.create({ url: authUrl });
+
+      return new Promise<auth.OAuth2User>((resolve, reject) => {
+        const handleUpdated = (
+          tabId: number,
+          changeInfo: browser.Tabs.OnUpdatedChangeInfoType,
+          tab: browser.Tabs.Tab
+        ) => {
+          if (changeInfo.status === "complete" && tabId === oAuthTab.id) {
+            const authorizationCode = this.extractCodeFromTabUrl(tab.url);
+
+            if (authorizationCode) {
```

Comment:
can you handle the `else` case or maybe better, reverse this check

`if (!authorizationCode) { throw new Error("...") }`

Reply:
we can do that. but the first ever event listener will trigger for original authURL as it's valid http url but we don't get any code there and for the second time after completiing auth flow we will have redirecturl tab where we get code.

so it will throw error for the first case unnecessarily. is it ok for you?

Reply:
Can we throw an error if there is no authorization code and the URL is the callback URL, but if it's the original auth URL we do not need that check. What do you think?

Reply:
on the other way, it seems fine to me as well. instead of introducing one more additional check just throw error if listener is triggered and no code is returned

Reply:
But then you mentioned we would throw an error when we should not, right? I don't think we should do that.

I just think we need to add error handling when we expect a code but one is not returned.